### ISSUE TITLE
Add a means to disregard a feedback's impact on score without discarding it

### ISSUE THREAD

**Issue by philipbelesky** [TabbycatDebate/tabbycat#538]
Currently if you want to remove a piece of feedback by discarding it then it still shows up on the feedback missing page. It was cited at Mexico as desirable to have an option to stop a piece of feedback impacting on a score but still keep it around. This could be a boolean type field whereby the score would then filter to exclude either `discarded` or `disregarded`.

**Comment by tienne-B** [comment]
Have a prototype for this, using a new `ignored` field (not either of the aforementioned names). (The pull request) How is feedback discarded currently? (Other than in the db view...) If there is such a place which I didn't find, the `ignored` setting could also be added to that view.

**Comment by czlee** [comment]
It's kind of not, we originally had plans to provide a means of discarding feedback easily, and never really implemented it. So the only way to discard it currently is in the db view.

In fact, more generally, in around 2014 we had much more ambitious plans for feedback, and dropped most of them as being too niche or too much effort for too little reward. Personally I think it's better for feedback to be spun out to a new system and worked on by a different team, so they can do it properly, but for the timebeing we're still doing improvements that are (hopefully) small but high-impact (like this one).

Keep the questions coming 🙂 As you might see I'm much better at answering questions quickly than reviewing pull requests (sorry, should free up a bit this coming week).

**Comment by philipbelesky** [comment]
The PR looks great, however I think at this point we should clarify a few things after that is merged but before closing this issue:

- [ ] It's probably worth noting (briefly) in the relevant docs page how to mark things as unconfirmed or ignored and what those mean specifically
- [ ] The interface text (on a feedback card) and the `help_text` for each field should probably explicitly call out the difference between them. In the PR the new interface text for ignored is "Ignored; will not affect this adjudicator's score" while the text for unconfirmed is "Ignored; will not affect this adjudicator's score" although the latter does also have a tooltip. Adding a tooltip to the former would help clarify what each means.
- [ ] Maybe a test or two just so our future selves need to worry less about messing with any complex interactions between ignored / unconfirmed on both scores and the missing feedback page

**Comment by tienne-B** [comment]
Add this point too:
* [x] Add method to easily mark feedback as being ignored :) I added a toggle link to do that in a new commit (same PR)

I also did some stuff for the first two points in that new commit, but the text could perhaps use some finessing (I myself do not feel that I fully understand the discarded/confirmed stuff)

**Comment by philipbelesky** [comment]
Thanks for adding the tests in #673 @tienne-B! If you really want to dig in to testing, checking how each flag affects scores could be useful, although definitely understand that testing is not the most appealing side of things.

I'll leave the issue open in the short term and might make some quick changes tonight or tomorrow to how the ignored/discarded state is set from the card-view before closing.

**Comment by philipbelesky** [comment]
Hey I've had a quick tinker with this. Changes/notes:

- This view is definitely something we wanted to restrict only to admins, so it needed the `AdministratorMixin` mixin to ensure it is authenticated
- Because it is admin only there is no need for the `AdminIgnoreFeedbackView` (or for the `Admin` prefix in the main view)
- I've switched to using a POST request rather than a GET as described above and so used `PostOnlyRedirectView` and reimplemented the link as a sneaky hidden form/button
- I've switched from getting the object directly rather than SingleObjectMixin; to me its make the code slightly more readable as it shifts the semantics from the class variables into the more explicit query / logic
- Split a long line used to constructed the message into a more clear if / else block (personal preference)
- Changed wording of 'recognise' to 'un-ignore' in the UI which is more clumsy but I think more explicit as it matches the wording used in the header element and the other 'ignore' links
- Added tooltip to the ignore/unignore link so its clear what it does if users have't already been used an ignored piece of feedback
- Added changelog description + credits

Let me know if anythings unclear or if you had any other feedback; otherwise we can consider the Issue closed. Thanks again!
